evm/src/cpu/kernel/asm/exp.asm

```diff
@@ -0,0 +1,70 @@
+global exp:
+// we don't seem to handle global labels yet, so this function has a local label too for now:
+exp:
```

Comment:
Could probably do with a simple comment along the lines of "Implementation using recursive expression `exp(x, e) = exp(x, e % 2) * exp(x * x, e / 2)`."

Reply:
I wrote out the pseudocode for the function, which should hopefully help :)